src/main.cu

```diff
@@ -99,35 +98,25 @@ __global__ void render(uint8_t *fb, int max_x, int max_y,int sample_per_pixel, c
  * @param objects Array of objects loaded from .obj file
  * @param number_of_meshes Number of objects in objects array 
  */
-__global__ void create_world(hitable **d_list, hitable **d_world,camera **d_camera, object3d **objects, int number_of_meshes) {
+__global__ void create_world(hitable **d_list, hitable **d_world,camera **d_camera, triangle *triangles, int number_of_meshes) {
     if (threadIdx.x == 0 && blockIdx.x == 0) {
         // TODO: Set the materials accordingly to the object
-        // For example create a property on object3d and an enum for the material type
-        // We can introduce a custom texture loader to load materials for the objects (bounded to our defined materials)
         material *mat = new lambertian(make_float3(0.5, 0.5, 0.5));
         // material *mat = new metal(make_float3(0.8, 0.6, 0.2), 0.0);
         // material *mat = new dielectric(1.5);
         
         int face_counter = 0;
 
         for (int i = 0; i < number_of_meshes; i++) {
-            for (int j = 0; j < objects[i]->num_triangles; j++) {
-                d_list[face_counter] = new triangle(objects[i]->triangles[j].v0, objects[i]->triangles[j].v1, objects[i]->triangles[j].v2, mat);
-                face_counter++;
-            }
+            d_list[i] = new triangle(triangles[i].v0, triangles[i].v1, triangles[i].v2, mat);

```

Comment:
Any ideas to make this better? I've tried:
`d_list[i]=&triangles[i]`
but this results in a error...